pmd-java/src/main/java/net/sourceforge/pmd/lang/java/oom/Metrics.java

```diff
@@ -123,7 +124,7 @@ public static double get(OperationMetricKey key, ASTMethodOrConstructorDeclarati
      * @return The value of the metric, or {@code Double.NaN} if the value couln't be computed or {@code option} is
      * {@code null}
      */
-    public static double get(OperationMetricKey key, ASTAnyTypeDeclaration node, ResultOption option) {
+    public static double get(MetricKey<OperationMetric> key, ASTAnyTypeDeclaration node, ResultOption option) {
```

Comment:
It's worth pointing out that these `get` overloads are working because of the use of _different_ 2nd, 3rd, etc. argument  types.  This method pattern may run into issues if a `ClassMetric` and `OperationMetric` needed the _same_ node and supplementary argument types due to type erasure on the `MetricKey` argument.